arangod/IResearch/IResearchFilterFactory.cpp

```diff
@@ -1316,7 +1416,25 @@ class ByTermSubFilterFactory {
         termFilter = &appendNot<irs::by_term>(
             append<irs::And>(*filter, filterCtx), filterCtx);
       } else {
-        termFilter = &append<irs::by_term>(*filter, filterCtx);
+        if (termsOpt.useByTerms) {
+          std::string name{filterCtx.query.namePrefix};
+          if (!nameFromAttributeAccess(name, *node.attribute, filterCtx.query,
+                                       filter != nullptr)) {
+            return {TRI_ERROR_BAD_PARAMETER,
+                    absl::StrCat("Failed to generate field name from node ",
+                                 aql::AstNode::toString(node.attribute))};
+          }
+          ScopedAqlValue value(*node.value);
+          if (!value.isConstant()) {
+            if (!value.execute(filterCtx.query)) {
+              // failed to execute expression
+              return {TRI_ERROR_BAD_PARAMETER, "could not execute expression"};
+            }
+          }
+          return appendByTermsFilter(filter, value, name, filterCtx, termsOpt);
+        } else {
```

Comment:
else is unnecessary

Reply:
Ah, I see, you mean because of return, not because of impossible to get here